gateway-js/src/index.ts

```diff
@@ -878,9 +878,11 @@ export class ApolloGateway implements GraphQLService {
 
       return getServiceDefinitionsFromRemoteEndpoint({
         serviceList,
-        ...(config.introspectionHeaders
-          ? { headers: config.introspectionHeaders }
-          : {}),
+        async getServiceIntrospectionHeaders(service) {
+          return typeof config.introspectionHeaders === 'function'
+            ? await config.introspectionHeaders(service)
```

Comment:
In the case this isn't an `async` function, it's no problem for us to `await` it anyway.

Reply:
(no action needed here, just pointing out this detail which becomes relevant once `gSIH` is optionally async)